device.yml

```diff
@@ -0,0 +1,234 @@
+%YAML 1.1
+---
+# yaml-language-server: $schema=https://raw.githubusercontent.com/harp-tech/reflex-generator/main/schema/device.json
+device: SyringePump
+whoAmI: 1280
+firmwareVersion: "1.0"
+hardwareTargets: "1.0"
+registers:
+  EnableMotorDriver: &enablefunctionreg
+    address: 32
+    access: Write
+    type: U8
+    maskType: EnableFlag
+    description: Enables the motor driver.
+  EnableProtocol:
+    <<: *enablefunctionreg
+    address: 33
+    description: Enables the currently defined protocol.
+  Step:
+    address: 34
+    access: [Write, Event]
+    type: U8
+    maskType: StepState
+    description: Status of the STEP motor controller pin.
+  Direction:
+    address: 35
+    access: [Write, Event]
+    type: U8
+    maskType: DirectionState
+    description: Status of the DIR motor controller pin.
+  ForwardSwitch: &switchreg
+    address: 36
+    access: Event
+    type: U8
+    maskType: ForwardSwitchState
+    description: Status of the forward limit switch.
+  ReverseSwitch:
+    <<: *switchreg
+    address: 37
+    maskType: ReverseSwitchState
+    description: Status of the reverse limit switch.
+  DigitalInput:
+    <<: *switchreg
+    address: 38
+    maskType: DigitalInputState
+    description: Status of the digital input pin.
+  DigitalOutputSet: &output
+    address: 39
+    type: U8
+    access: Write
+    maskType: DigitalOutputState
+    description: Set the specified digital output lines.
+  DigitalOutputClear:
+    <<: *output
+    address: 40
+    description: Clear the specified digital output lines.
+  DO0Sync: &ioconfig
+    address: 41
+    type: U8
+    access: Write
+    maskType: DO0SyncConfig
+    description: Configures which signal is mimicked in the digital output 0.
+  DO1Sync:
+    << : *ioconfig
+    address: 42
+    maskType: DO1SyncConfig
+    description: Configures which signal is mimicked in the digital output 1.
+  DI0Trigger:
+    << : *ioconfig
+    address: 43
+    maskType: DI0TriggerConfig
+    description: Configures the callback function triggered when digital input is triggered.
+  Microstepping:
```

Comment:
A qualifier still seems more readable here, e.g. `MicrostepConfig`. Also it feels from reading the mask values and other similar driver documentation that Full and Half, Quarter are still not considered "micro-stepping" by some.

Since the term "microstep" is not used anywhere else we could just simplify it to `StepConfig`.

Reply:
https://thepihut.com/products/sparkfun-easydriver-stepper-motor-driver?variant=39553135345859&currency=GBP&utm_medium=product_sync&utm_source=google&utm_content=sag_organic&utm_campaign=sag_organic&gclid=CjwKCAjw0ZiiBhBKEiwA4PT9zyKqJh2zmlpF4V-_bj3q3k9vhpQNJIGSYawOF-1WgmHOw6J8EnXZLBoCi_wQAvD_BwE

I have been trying to follow some of the nomenclature from sparkfun or affiliates. For instance:
"MS1 and MS2 pins broken out to change microstepping resolution to full, half, quarter and eighth steps (defaults to eighth)"

StepConfig i am not a huge fan, how about `StepResolution` and `StepResolutionConfig`?

Reply:
Microstep resolution is typically the "hardware" nomenclature for this.  StepResolution or StepResolutionConfig are ok for me.

Reply:
@artursilva0 @bruno-f-cruz I agree the suffix `Config` is not great. `Mode` is much more common when discussing stepper motor modes, it's called exactly the step mode of the motor, as in full-step mode, half-step mode, etc.

The reason I prefer `Mode` instead of `Resolution` in this case is because I think the names of values of the enum can be misleading (Full, Half, Quarter) etc. If the name of the register is `MicrostepResolution` and I have no idea and read `Full` and `Half` I might naively interpret `Full` to mean "Full resolution" instead of a "Full step" which is actually the smallest resolution (similar for Half which is Half-Step and not Half-resolution!).

`StepMode` also conveys more clearly for me that this is a discrete set of modes, and not actually a continuous number, `StepResolution` makes me think in steps per revolution.

Finally, using the suffix `Config` for a group mask doesn't sound quite right to me, especially when only one value can be set. I find `Config` more natural as a qualifier for objects rather than an enum type, but it might be just me.

Given this I would amend my proposal to:

register: `StepMode`
mask: `StepModeType`